### ISSUE TITLE
Embedded arrays, objects

### ISSUE THREAD

**Issue by jpmens** [jpmens/jo#19]
Just playing around, but I'd like some feedback (cc: @itchyny) Is this a good idea?

creates an array: `name[]`
creates an object: `name[member]`

```bash
jo -p number=17 pass=true geo[lon]=88 geo[cc]=ES point[]=1 point[]=2 geo[lat]=123.45 geo[ok]=false
```

```json
{
   "number": 17,
   "pass": true,
   "geo": {
      "lon": 88,
      "cc": "ES",
      "lat": 123.45,
      "ok": false
   },
   "point": [
      1,
      2
   ]
}
```

**Comment by tiptronic** [comment]
I like it!!

**Comment by phanimahesh** [comment]
:+1: This is a very common format, for instance url query parameters in many web frameworks are interpreted this way.

**Comment by jpmens** [comment]
Implemented in 0.8
